Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaSourceAVFObjC.mm

```diff
@@ -544,36 +544,40 @@ void getSupportedTypes(HashSet<String>& types) const final
         ALWAYS_LOG(LOGIDENTIFIER);
         MediaTime synchronizerTime = PAL::toMediaTime([m_synchronizer currentTime]);
 
+        // In cases where the destination seek time precisely matches the synchronizer's existing time
+        // no time jumped notification will be issued. In this case, just notify the MediaPlayer that
+        // the seek completed successfully.
         m_synchronizerSeeking = synchronizerTime != seekedTime;
         ALWAYS_LOG(LOGIDENTIFIER, "seekedTime = ", seekedTime, ", synchronizerTime = ", synchronizerTime, "synchronizer seeking = ", m_synchronizerSeeking);
 
         [m_synchronizer setRate:0 time:PAL::toCMTime(seekedTime)];
 
-        // In cases where the destination seek time precisely matches the synchronizer's existing time
-        // no time jumped notification will be issued. In this case, just notify the MediaPlayer that
-        // the seek completed successfully.
-        if (!m_synchronizerSeeking)
-            seekCompleted();
+        m_mediaSourcePrivate->seekToTime(seekedTime, [this, weakThis] {
+            if (!weakThis)
```

Comment:
I wonder whether we should not build a protectedThis here, given our current smart pointer policy approach. Ditto in other part of the patch where we have a weakThis in a lambda.

Reply:
Those aren't refcounted object.